text/0000-const-type-inference.md

```diff
@@ -36,12 +36,23 @@ amount of boilerplate reduces their mental load. This reduction in mental load a
 [guide-level-explanation]: #guide-level-explanation
 
 You may declare constants and static variables without specifying their types when the type can be inferred
-from the initial value. For example:
+from the initial value, subjecting to the following constraints:
+- All numerical types contributing to the inference must be specified.
```

Comment:
This description doesn't match the examples below. My reading of this would be that 

```rust
const PI: _ = 3.1415_f32
```
because it has an inferred type that is numeric. And as written, this rule feels wrong to me, because it treats numeric types specially.

How about something like:

 "the types of all literals must be fully constrained, which generally means numeric literals must either have a type suffix, or the type must specify their type"
 
 or
 
 "the type of any integer or floating point literal in the expression must have a type that is uniquely determine from the context"
 
 to mirror the wording from https://doc.rust-lang.org/reference/expressions/literal-expr.html